azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -41,11 +52,56 @@ def __init__(self):
         self.required_features = []
 
     @staticmethod
-    def from_extensions_config(xml_text):
-        extensions_goal_state = ExtensionsGoalState()
-        extensions_goal_state.xml_text = xml_text
+    def create_empty():
+        return ExtensionsGoalState()
+
+    @staticmethod
+    def create_from_extensions_config(incarnation, xml_text):
+        return _ExtensionsGoalStateFromExtensionsConfig(incarnation, xml_text)
+
+    @staticmethod
+    def create_from_vm_settings(etag, json_text):
+        return _ExtensionsGoalStateFromVmSettings(etag, json_text)
+
+    def get_id(self):
+        """
+        Returns the incarnation number if the ExtensionsGoalState was created from ExtensionsConfig, or the etag if it
+        was created from vmSettings.
+        """
+        return self._id
+
+    def get_redacted_text(self):
+        """
+        Returns the raw text (either the ExtensionsConfig or the vmSettings) with any confidential data removed.
+        """
+        return None
+
+    @staticmethod
+    def compare(from_extensions_config, from_vm_settings):
+        """
+        Compares the current instance against 'other' and logs a GoalStateMismatch message if they are different
```

Comment:
NIT: The comment suggests that the `self` will be compared with `other` but the function definition is a static class comparing 2 different arguments. 

Maybe modify the comment a bit to make it a bit more clearer?

Reply:
thanks, initially this was a member function and i made it static without updating the comment; will do that

Reply:
done